addons/calendar/static/src/views/calendar_form/calendar_form_controller.js

```diff
@@ -38,4 +39,25 @@ export class CalendarFormController extends FormController {
         }
         return super.beforeExecuteActionButton(...arguments);
     }
+
+    /**
+     * Custom delete function for calendar events.
+     * it performs the default delete. Otherwise, it calls the unlink action on the server.
+     */
```

Comment:
To be updated:
```suggestion
    /**
     * Custom delete function for calendar events, which can call the unlink action or not.
     * When << Insert condition here >>, it performs the default delete.
     * Otherwise, it calls the unlink action on the server.
     */
```